packages/state-transition/src/util/fulu.ts

```diff
@@ -13,7 +15,14 @@ export function initializeProposerLookahead(state: CachedBeaconStateElectra): Va
   const lookahead: ValidatorIndex[] = [];
 
   for (let i = 0; i < MIN_SEED_LOOKAHEAD + 1; i++) {
-    lookahead.push(...computeProposerIndices(ForkSeq.fulu, state, currentEpoch + i));
+    const epoch = currentEpoch + i;
+
+    const shuffling =
+      state.epochCtx.getShufflingAtEpochOrNull(epoch) ??
+      // Shuffling might not be available for next epoch
+      computeEpochShuffling(state, getActiveValidatorIndices(state, epoch), epoch);
```

Comment:
this code's purpose is really the same to `packages/state-transition/src/epoch/processProposerLookahead.ts` so I prefer to do the same pattern for both to avoid concern for future

we can also extract to a separate method in `EpochCache` and handle fallback there, instead of at 2 places

Reply:
> we can also extract to a separate method in EpochCache and handle fallback there, instead of at 2 places

nvm, it's not convenient to do that

I realize that it's super slow to do `getActiveValidatorIndices()` in the for loop
  - if `i == 0` we should get active indices from `currentShuffling` of `EpochCache`
  - if `i == 1` we should get active indices from `EpochTransitionCache.nextShufflingActiveIndices`
  - otherwise call `getActiveValidatorIndices()`. We never reach this with the current spec.